app/src/org/commcare/dalvik/activities/AppManagerActivity.java

```diff
@@ -0,0 +1,169 @@
+package org.commcare.dalvik.activities;
+
+import org.commcare.android.adapters.AppManagerAdapter;
+import org.commcare.android.util.SessionUnavailableException;
+import org.commcare.dalvik.R;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.dalvik.services.CommCareSessionService;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.AdapterView.OnItemClickListener;
+import android.widget.ListView;
+import android.widget.Toast;
+
+/**
+ * The activity that starts up when a user launches into the app manager.
+ * Displays a list of all installed apps, each of which can be clicked to launch
+ * the SingleAppManagerActivity for that app. Also includes a button for
+ * installing new apps.
+ *
+ * @author amstone326
+ */
+
+public class AppManagerActivity extends Activity implements OnItemClickListener {
```

Comment:
I guess so, yes. Neither AppManagerActivity nor SingleAppManagerActivity make use of any of the framework or methods provided by CommCareActivity, so I wasn't sure if was necessary or proper. Especially since they are also distinctly not part of the essential CommCare "app" and don't really follow the same workflows. What are the pros of doing so?

Also, not sure what you mean by the clean up comment, since there are only 2 lines of code dealing with that.